src/scripts/components/main/onboarding/onboarding.tsx

```diff
@@ -0,0 +1,147 @@
+import React, {FunctionComponent, useEffect, useState} from 'react';
+import {useIntl} from 'react-intl';
+import cx from 'classnames';
+
+import Button from '../button/button';
+import {CloseIcon} from '../icons/close-icon';
+
+import styles from './onboarding.styl';
+
+interface Props {
+  id: number;
+  onPageChange: (id: number) => void;
+  onClose: () => void;
+}
+
+const Onboarding: FunctionComponent<Props> = ({id, onPageChange, onClose}) => {
+  const intl = useIntl();
+
+  const onboardingContent = [
+    {
+      id: 1,
+      content: intl.formatMessage({id: 'tooltip-stories'}),
+      elementId: 'stories'
+    },
+    {
+      id: 2,
+      content: intl.formatMessage({id: 'tooltip-layers'}),
+      elementId: 'layers'
+    },
+    {
+      id: 3,
+      content: intl.formatMessage({id: 'tooltip-share'}),
+
+      elementId: 'share'
+    },
+    {
+      id: 4,
+      content: intl.formatMessage({id: 'tooltip-menu'}),
+      elementId: 'menu'
+    },
+    {
+      id: 5,
+      content: intl.formatMessage({id: 'tooltip-projection'}),
+      elementId: 'projection'
+    },
+    {
+      id: 6,
+      content: intl.formatMessage({id: 'tooltip-compass'}),
+      elementId: 'compass'
+    },
+    {
+      id: 7,
+      content: intl.formatMessage({id: 'tooltip-download'}),
+      elementId: 'download'
+    }
+  ];
+  const currentOnboarding = onboardingContent.find(
+    content => content.id === id
+  );
+
+  const referenceElement = document.querySelector(
+    `#${currentOnboarding?.elementId}`
+  );
+
+  const [referencePosition, setReferencePosition] = useState<DOMRect | null>(
+    null
+  );
+
+  // set position of onboarding tooltip
+  useEffect(() => {
```

Comment:
Why do you use the useEffect here? 
It think it should work with something like 
```
  const rect = referenceElement?.getBoundingClientRect() || null;
  const [referencePosition, setReferencePosition] = useState(rect);
```
and if it should only run when the element changes useMemo is the best hook here: 
```
  const rect = useMemo(() => referenceElement?.getBoundingClientRect() || null, [referenceElement]);
  const [referencePosition, setReferencePosition] = useState(rect);
```